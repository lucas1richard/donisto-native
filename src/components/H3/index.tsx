import * as React from 'react';
import { Text } from 'react-native';
import styles from './styles';

export interface IH3Props {
  children: string;
  color?: 'primary'|'secondary'|'default'|'light'
}

const H3 = ({ children, color }: IH3Props) => {
  return (
    <Text style={styles[color]}>
      {children}
    </Text>
  );
};

H3.defaultProps = {
  color: 'default'
};

export default H3;
