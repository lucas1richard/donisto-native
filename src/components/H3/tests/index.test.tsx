import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import H3, { IH3Props } from '..';

describe('<H3 />', () => {
  const renderComponent = (props: IH3Props) => toJson(shallow(<H3 {...props} />));
  it('renders with just text', () => {
    expect(renderComponent({ children: 'mockText' })).toMatchSnapshot();
  });
  it('renders with a color prop', () => {
    expect(renderComponent({
      children: 'mockText',
      color: 'light',
    })).toMatchSnapshot();
  });
});