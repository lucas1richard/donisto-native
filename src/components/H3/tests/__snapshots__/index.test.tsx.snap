// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H3 /> renders with a color prop 1`] = `
<Component
  style={
    Object {
      "color": "#fff",
      "fontSize": 28,
    }
  }
>
  mockText
</Component>
`;

exports[`<H3 /> renders with just text 1`] = `
<Component
  style={
    Object {
      "color": "#000000",
      "fontSize": 28,
    }
  }
>
  mockText
</Component>
`;
